iscsidev/iscsi.go

```diff
@@ -48,9 +49,17 @@ type Device struct {
 	BSOpts      string
 
 	targetID int
+
+	nsexec *lhns.Executor
 }
 
 func NewDevice(name, backingFile, bsType, bsOpts string, scsiTimeout, iscsiAbortTimeout int64) (*Device, error) {
+	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
+	nsexec, err := lhns.NewNamespaceExecutor(util.ISCSIdProcess, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Please make sure `nsexec` thread safe, if multiple methods using it will be called concurrently.

Reply:
the `nsexec` wraps `nsenter` around the command, then proceeds to invoke the `os/exec` `cmd.Run()` method, initiating an external process. This is the same as the current implementation but replaced the execution from the host namespace to the iscsid namespace.